linera-chain/src/certificate.rs

```diff
@@ -69,13 +75,25 @@ impl<T: Debug> Debug for GenericCertificate<T> {
     }
 }
 
-// NOTE: For backwards compatiblity reasons we serialize the new `Certificate` type as the old
-// one. Otherwise we would be breaking the RPC API schemas. We can't implement generic serialization
-// for `Certificate<T>` since only specific `T`s have corresponding `CertificateValue` variants.
 impl Serialize for ValidatedBlockCertificate {
     fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
-        let cert = Certificate::from(self.clone());
-        cert.serialize(serializer)
+        let mut state = serializer.serialize_struct("ValidatedBlockCertificate", 4)?;
+        state.serialize_field("hash", &self.value.hash())?;
+        state.serialize_field("value", &self.inner())?;
+        state.serialize_field("round", &self.round)?;
+        state.serialize_field("signatures", &self.signatures)?;
+        state.end()
+    }
+}
+
+impl Serialize for TimeoutCertificate {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
```

Comment:
It's maybe safer to destructure `self` here. Then if we add a field, the compiler reminds us to also serialize it.

Reply:
Done in https://github.com/linera-io/linera-protocol/pull/2845/commits/cdb684a37b4fe914ecf9a4fd9c2599e6ef97a6aa.